### PR TITLE
[CIFuzz][NFC] Add skeleton for ClusterFuzzLite and Filestore

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Add methods that will be used for ClusterFuzzLite and Filestore but do not implement them.

### CODE REVIEW

File: infra/cifuzz/filestore/__init__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# pylint: disable=unused-argument,no-self-use
+class BaseFilestore:
+  """Base class for a filestore."""
+
+  def __init__(self, config):
+    self.config = config
+
+  def upload_corpus(self, name, directory):
+    """Uploads the corpus located at |directory| to |name|."""
+    NotImplementedError('Child class must implement method.')
```
Comment (oliverchang) [inline infra/cifuzz/filestore/__init__.py:26]: you're missing raise here.

File: infra/cifuzz/clusterfuzz_deployment.py [hunk 2]
```diff
@@ -92,14 +133,19 @@ def get_latest_build_name(self):
...
+  def download_latest_build(self, parent_dir):
     """Downloads the latest OSS-Fuzz build from GCS.
 
     Returns:
       A path to where the OSS-Fuzz build was stored, or None if it wasn't.
     """
-    build_dir = os.path.join(out_dir, self.BUILD_DIR_NAME)
+    build_dir = self.get_build_dir(parent_dir)
     if os.path.exists(build_dir):
+      # This path is necessary because download_latest_build can be called
```
Comment (oliverchang) [inline infra/cifuzz/clusterfuzz_deployment.py:144]: this comment is still a bit confusing to me, can this just be shortened to something along the lines of "this function may be called multiple times, don't repeat the download in such cases" ?

**Review by oliverchang — APPROVED** [review]

File: infra/cifuzz/filestore/__init__.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/cifuzz/filestore/__init__.py:26]: Good call. Thanks for pointing this out.

File: infra/cifuzz/clusterfuzz_deployment.py (hunk 2 above)
Comment (jonathanmetzman) [reply infra/cifuzz/clusterfuzz_deployment.py:144]: Done.

**Pull request merged**